### DISCUSSION TITLE
A way to check if an untrusted std::string can be safely parsed to JSON

### DISCUSSION THREAD

**Discussion by cpiernikowski** [nlohmann/json#2769] [Q&A] [answered]
Im currently dealing with an untrusted input and I need a way to check if a certain std::string can be converted to a JSON object.

```
std::string str = getUntrustedStr();
if (canBeParsed(str))
     nlohmann::json jsonObj = nlohmann::json::parse(str);
```
Is there any way to achieve something similar to this?
Should i go with try and catch?

**Comment by nlohmann** [comment, marked as answer]
See https://json.nlohmann.me/api/basic_json/accept/

**Reply by cpiernikowski** [reply]
Thank you, that's exactly what i was looking for :)
